Development/src/pages/nodes/NodesShow.js

```diff
@@ -0,0 +1,159 @@
+    useShowController,
+} from 'react-admin';
+import Cookies from 'universal-cookie';
+import LinkIcon from '@material-ui/icons/Link';
+import ItemArrayField from '../../components/ItemArrayField';
+import JsonIcon from '../../components/JsonIcon';
+import MapTags from '../../components/TagsField';
+import TAIField from '../../components/TAIField';
+import UrlField from '../../components/URLField';
+import QueryVersion from '../../components/QueryVersion';
+import ChipConditionalLabel from '../../components/ChipConditionalLabel';
+
+const cookies = new Cookies();
+
+const NodesTitle = ({ record }) => {
+    return (
+        <span>
+            Node:{' '}
+            {record
+                ? record.label
+                    ? `${record.label}`
+                    : `${record.id}`
+                : 'Unknown'}
+        </span>
+    );
+};
+
+function buildLink(record) {
+    return record.protocol + '://' + record.host + ':' + record.port;
+}
+
+const NodesShowActions = ({ basePath, data, resource }) => (
+    <TopToolbar title={<NodesTitle />}>
+        {data ? (
+            <Button
+                label={'Raw'}
+                href={cookies.get('Query API') + '/' + resource + '/' + data.id}
+                title={'View raw'}
+            >
+                <JsonIcon />
+            </Button>
+        ) : null}
+        <ListButton title={'Return to ' + basePath} basePath={basePath} />
+    </TopToolbar>
+);
+
+export const NodesShow = props => {
+    const controllerProps = useShowController(props);
+    return (
+        <ShowView
+            {...props}
+            {...controllerProps}
+            title={<NodesTitle />}
+            actions={<NodesShowActions />}
+        >
+            <SimpleShowLayout>
+                <TextField label="ID" source="id" />
+                <TAIField source="version" />
+                <TextField source="label" />
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <TextField source="description" />
+                )}
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <FunctionField
+                        label="Tags"
+                        render={record =>
+                            Object.keys(record.tags).length > 0
+                                ? MapTags(record)
+                                : null
+                        }
+                    />
+                )}
+                <hr />
+                <UrlField source="href" label="Address" />
+                <TextField source="hostname" />
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <ItemArrayField
+                        label="API Versions"
+                        source="api.versions"
+                    />
+                )}
+                {controllerProps.record && QueryVersion() >= 'v1.1' && (
+                    <ArrayField label=" API Endpoints" source="api.endpoints">
```

Comment:
Why _is_ there an extra space at the start of the label? (I see it was there already!)